src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -96,29 +99,101 @@ public static Email parseEmail(String email) throws ParseException {
+    public static Offer parseOffer(String offer) throws ParseException {
+        requireNonNull(offer);
+        String trimmedOffer = offer.trim();
+
+        if (!trimmedOffer.contains(" ")) {
+            throw new ParseException(Offer.MESSAGE_CONSTRAINTS);
+        }
+
+        String[] goodPricePair = splitOnLastWhitespace(trimmedOffer);
+
+        String goodString = goodPricePair[0];
+        String priceString = goodPricePair[1];
+        GoodName good = parseGood(goodString);
+        Price price = parsePrice(priceString);
+        return new Offer(good, price);
+    }
+
+    /**
+     * Parses a {@code String good} into a {@code Good}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code good} is invalid.
+     */
+    public static GoodName parseGood(String good) throws ParseException {
+        requireNonNull(good);
+        String trimmedGood = good.trim();
+        if (!GoodName.isValidGoodName(trimmedGood)) {
+            throw new ParseException(GoodName.MESSAGE_CONSTRAINTS);
+        }
+        return new GoodName(trimmedGood);
+    }
+
+    /**
+     * Parses a {@code String price} into a {@code price}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code price} is invalid.
+     */
+    public static Price parsePrice(String price) throws ParseException {
+        requireNonNull(price);
+        String trimmedPrice = price.trim();
+        if (!Price.isValidPrice(trimmedPrice)) {
+            throw new ParseException(Price.MESSAGE_CONSTRAINTS);
         }
-        return new Tag(trimmedTag);
+
+        return new Price(trimmedPrice);
     }
 
     /**
-     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     * Parses {@code Collection<String> offers} into a {@code Set<Offer>}.
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static Set<Offer> parseOffers(Collection<String> offers) throws ParseException {
+        requireNonNull(offers);
+        final Set<Offer> offerSet = new HashSet<>();
+        for (String offer : offers) {
+            offerSet.add(parseOffer(offer));
         }
-        return tagSet;
+        return offerSet;
+    }
+
+    /**
+     * Returns a {@code String} array as if {@code String.split()} is called only on its last whitespace.
+     * Assumes: the {@code String} is already stripped of trailing and leading whitespaces,
+     * and contains at least one whitespace.
+     *
+     * @param string the {@code String} to be split
+     * @return the {@code String} array containing the split result
+     */
+    public static String[] splitOnLastWhitespace(String string) {
+        String[] words = string.split(" ");
```

Comment:
Yes, a requireNonNull is necessary since it is called outside the class. But, splitting on whitespace without adding a -1 flag will trim the words already